File: internal/models/log.go
package models

import (
	"time"
)
type LogEvent struct{
	ID          string			`json:"id"`
	TimeStamp   time.Time		`json:"time"`
	Level       string			`json:"level"`
	Service     string			`json:"service"`
	Message     string 			`json:"message"`
}

File: frontend/src/types/log.ts
export type LogLevel = "INFO" | "DEBUG" | "WARN" | "ERROR";

export interface LogEvent {
  id: string;
  time: string;
  level: LogLevel;
  service: string;
  message: string;
}

export type ConnectionState = "connecting" | "connected" | "disconnected";
